File: integration/python/examples/build_redeem.py
from __future__ import annotations

import argparse
import sys

from chancery_integration import (
    build_redeem_direct_instruction,
    load_redeem_direct_operation,
    serialize_instruction,
)


def main() -> None:
    parser = argparse.ArgumentParser(
        description="Build a Chancery redeem_direct instruction document."
    )
    parser.add_argument("operation_file")
    arguments = parser.parse_args()

    operation = load_redeem_direct_operation(arguments.operation_file)
    instruction = build_redeem_direct_instruction(operation)
    sys.stdout.write(serialize_instruction(instruction))


if __name__ == "__main__":
    main()

File: integration/typescript/examples/build_redeem.ts
import {
    buildRedeemDirectInstruction,
    loadRedeemDirectOperation,
    serializeInstruction,
} from "../src/index.ts";

const operationFile = process.argv[2];
if (operationFile === undefined) {
    throw new Error("Usage: build_redeem.ts <direct-redeem.operation.json>");
}

const operation = loadRedeemDirectOperation(operationFile);
const instruction = buildRedeemDirectInstruction(operation);
process.stdout.write(serializeInstruction(instruction));
